snuba/snuba_migrations/spans/0007_spans_add_measure_hashmap.py

```diff
@@ -0,0 +1,75 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Array, Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+
+storage_set_name = StorageSetKey.SPANS
+local_table_name = "spans_local"
+dist_table_name = "spans_dist"
+
+# There an issue in Clickhouse where the arrayMap function passes
+# in Nothing type values for empty arrays. This causes the regex function to fail
+# without the toString function, unless a merge for the part is completed.
+# This is fixed in Clickhouse 22.
+MEASUREMENTS_HASH_MAP_COLUMN = (
+    "arrayMap((k) -> "
+    "cityHash64(replaceRegexpAll(toString(k), '(\\\\=|\\\\\\\\)', '\\\\\\\\\\\\1')),"
+    "measurements.key)"
+)
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Adds measurements tags hash map column defined as Array(Int64)
+    Materialized with MEASUREMENTS_HASH_MAP_COLUMN expression.
+    This allows us to quickly find measure tag keys and we can add an
+    index on this column. Unlike sentry tags, we don't hash the
+    tag values since we don't need to query on them.
```

Comment:
Why doesn't starfish want to filter based on the tag values?

Reply:
I am assuming you meant the measurements value here? If so, the measurements value is a float64 column and finding exact matches on float values is unnecessary. The use case right now is for finding spans with specific measurement keys. Also, most likely the use case is going to be finding measurement values with range checks for which the hash map column would not be useful.